### PR TITLE
Refactor User Interface

### PR DESCRIPTION
**Pull request by LeNitrous**
First and foremost, this is the awaited UI refresh which now sports a sidebar instead of a full screen menu. This also sports updated styling on several components and updates osu!framework and Fluent System Icons. Backdrops (backgrounds) get a significant update as well now allowing video and images as a target.

Under the hood, I have refactored themeing and keybind management (UI is again to follow). Themes can now be edited on the fly but only the export button works. Applying live will follow. I've also laid down the foundation to avatar, recognition, and camera settings but only as hollow controls that don't do anything.

### CODE REVIEW

**Comment by sr229** [comment]
Have you verified the new theme is satisfactory for this refactor?

**Comment by LeNitrous** [comment]
Not yet. There's a few more things missing as well.

**Comment by Speykious** [comment]
Forgot to review the UI changes yesterday... I guess I will today then.

**Comment by adryzz** [comment]
When you open a sub-menu, like the debug settings, you can only go back using the back arrow.

That is fine, but if the categories of the previous menu (icons on the right) are still visible (even if disabled), one will try clicking them instead to go back.

image

So, either hiding them or making it so you go back when you click that area is a better idea, for the user's perspective.

Also, i don't know if this is known or not, but the text box control (and i assume all the controls) does not support unicode. I'll look more into this to see if it was an issue before this PR, but i discovered it now.

image

The rest looks good to me

**Comment by LeNitrous** [comment]
@adryzz
1. Made it so clicking the navigation buttons closes all open sub panels and scrolls to the desired section.
2. That's because the font doesn't really support unicode unfortunately.
3. The playback safely stops on Windows, seems to be crashing for Linux. Should be fixed.
@sr229 @Speykious will need your reviews on this

**Comment by adryzz** [comment]
> 1. Made it so clicking the navigation buttons closes all open sub panels and scrolls to the desired section.
> 2. That's because the font doesn't really support unicode unfortunately.
> 3. The playback safely stops on Windows, seems to be crashing for Linux. Should be fixed.

Tested all 3. Looks good so far.

**Comment by LeNitrous** [comment]
Just a note, I added file extension filters to only support PNG and JPG for images while MP4 for videos. It is possible to load in other formats and can be supported by the framework, but for sanity's sake, those are the defaults for now until there is demand for others.

**Comment by sr229** [comment]
Keep in mind I am unavailable for two weeks while my residence is currently evaluating how much repairs will be needed due to the typhoon. Only merge if the current participants are satisfied with the changes.

**Comment by adryzz** [comment]
> Just a note, I added file extension filters to only support PNG and JPG for images while MP4 for videos. It is possible to load in other formats and can be supported by the framework, but for sanity's sake, those are the defaults for now until there is demand for others.

MKV videos don't work framework side from my testing, tried with H264 and HEVC, but [they are working on adding hardware accelerated video playback to o!f](https://github.com/ppy/osu-framework/issues/4079) anyway (at least for Windows and Mac... VAAPI is hard to use)

**Comment by Speykious** [comment]
Tested the background feature, and it's great!

I clicked on "Open theme designer" and the app crashed, I don't know if that's expected.
[vignette.stdout.log](https://github.com/vignetteapp/vignette/files/7128881/vignette.stdout.log)
[vignette.stderr.log](https://github.com/vignetteapp/vignette/files/7128883/vignette.stderr.log)

**Comment by Speykious** [comment]
The light theme still lacks contrast in some places imo.
I highlighted the good contrast in green, the bad ones in red, and the one I'm unsure of in yellow.
image

The dark theme is almost perfect, except for the placeholder text from the search bar. I think if it had the same wolor as the subtitles like `General` down there the theme would become very solid.
image

**Review by adryzz — DISMISSED** [review]

**Review by Speykious — CHANGES_REQUESTED** [review]
I can't tell where exactly in the code, but Vignette doesn't remember the proper size of the window when it's resizable. It seems to take the resolution from the resolution config instead.
I always make my window maximized, yet each time I open Vignette, it opens it like this:
image

All good besides that though.

**Comment by adryzz** [comment]
@Speykious it does work on my end

**Comment by Speykious** [comment]
> @Speykious it does work on my end

For me, it remembers whether it is resizable or not, but not the resolution when in resizable mode.

**Comment by Speykious** [comment]
TL;DR: there's a resizable bug but it can be addressed in a separate pull request. For now, when it comes to the design, the dark mode is almost perfect, but the light mode requires a bit more work to be comfortable.

**Review by Speykious — DISMISSED** [review]

**Comment by sr229** [comment]
> TL;DR: there's a resizable bug but it can be addressed in a separate pull request. For now, when it comes to the design, the dark mode is almost perfect, but the light mode requires a bit more work to be comfortable.

This is only within the limitations of the current Fluent theme designer, we can't change the colors anymore since it already is within the acceptable contrast guidelines,

**Comment by sr229** [comment]
> I can't tell where exactly in the code, but Vignette doesn't remember the proper size of the window when it's resizable. It seems to take the resolution from the resolution config instead.
> I always make my window maximized, yet each time I open Vignette, it opens it like this:
> image
> 
> All good besides that though.

That's something we cannot fix since the last saved resolution is provided by your WM, not us.

**Comment by LeNitrous** [comment]
Should be ready for review. Keep in mind that this is about the side panel itself. Specifics such as Live2D, Backdrop, Themeing and other features will have to be addressed in separate PRs.

**Review by sr229 — APPROVED** [review]

**Comment by Speykious** [comment]
> > I can't tell where exactly in the code, but Vignette doesn't remember the proper size of the window when it's resizable. It seems to take the resolution from the resolution config instead.
> > I always make my window maximized, yet each time I open Vignette, it opens it like this:
> > image
> > All good besides that though.
> 
> That's something we cannot fix since the last saved resolution is provided by your WM, not us.

Oh... Yeah that makes sense
